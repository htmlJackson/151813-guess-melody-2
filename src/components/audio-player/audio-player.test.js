import React from "react";
import renderer from "react-test-renderer";

import AudioPlayer from "../audio-player/audio-player.jsx";

jest.mock(`./audio-player`);

describe(`Проверка <AudioPlayer>`, () => {
  jest.spyOn(window.HTMLMediaElement.prototype, `play`).mockImplementation(() => {});
  jest.spyOn(window.HTMLMediaElement.prototype, `pause`).mockImplementation(() => {});
  it(`компонент <AudioPlayer> корректно отрисован`, () => {
    const mocSrc = `https://patrickdearteaga.com/audio/Child's%20Nightmare.ogg`;

    const tree = renderer.create(<AudioPlayer isPlaying={false} onPlayButtonClick={jest.fn()} src={mocSrc}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
