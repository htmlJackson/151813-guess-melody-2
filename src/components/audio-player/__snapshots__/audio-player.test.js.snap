// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка <AudioPlayer> компонент <AudioPlayer> корректно отрисован 1`] = `null`;
